test/mako/alerter/slack/message.go

```diff
@@ -65,16 +65,19 @@ func (smh *messageHandler) Post(text string) error {
 	// TODO(Fredy-Z): add deduplication logic, maybe do not send more than one alert within 24 hours?
 	errs := make([]error, 0)
 	channels := smh.config.channels
+	mux := &sync.Mutex{}
 	var wg sync.WaitGroup
 	for i := range channels {
 		channel := channels[i]
 		wg.Add(1)
 		go func() {
+			defer wg.Done()
 			message := fmt.Sprintf(messageTemplate, time.Now(), text)
 			if err := smh.client.Post(message, channel.identity); err != nil {
-				errs = append(errs, fmt.Errorf("Failed to send message to channel %v", channel))
+				mux.Lock()
```

Comment:
I guess this is not high perf code, but you can just create a channel and send errors there and consume them on the other side after `wg.Wait`. But this is fine for this.